pkg/controller/endpointslice/utils.go

```diff
@@ -0,0 +1,257 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package endpointslice
+
+import (
+	"bytes"
+	"crypto/md5"
+	"encoding/hex"
+	"fmt"
+	"hash"
+	"reflect"
+	"sort"
+	"time"
+
+	"github.com/davecgh/go-spew/spew"
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/api/core/v1"
+	discovery "k8s.io/api/discovery/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/klog"
+	podutil "k8s.io/kubernetes/pkg/api/v1/pod"
+	api "k8s.io/kubernetes/pkg/apis/core"
+	hashutil "k8s.io/kubernetes/pkg/util/hash"
+)
+
+func podChanged(oldPod, newPod *corev1.Pod) bool {
```

Comment:
consider refactor the original these function from `pkd/controller/endponit` into a separate package and share it. Maybe `pkd/controller/util/utils.go`

Reply:
```
func PodChanged(oldPod, newPod *corev1.Pod, func(oldPod, newPod *corev1.Pod) bool) 


func(controller endpointSliceController) isPodEndpointDifferent(oldPod, newPod *corev1.Pod) bool{
        newEndpoint := podToEndpoint(newPod, &corev1.Node{}, false)
	oldEndpoint := podToEndpoint(oldPod, &corev1.Node{}, false)
      .....
    
}
```